packages/core/src/types.ts

```diff
@@ -704,3 +667,16 @@ export type IncomingMessage = {
     payload: INCOMING_EVENT_PAYLOADS[K];
   };
 }[keyof INCOMING_EVENT_PAYLOADS];
+
+type StructuralEditorModeProps =
+  | {
+      editorMode: true;
+      node: ExperienceTreeNode;
+    }
+  | {
+      editorMode?: false;
+    };
+
+export type StructureComponentProps<OtherProps> = React.PropsWithChildren<
+  StructuralEditorModeProps & OtherProps
+>;
```

Comment:
nit: not a big fan of these implicit generics, it could be just `StructuralEditorModeProps` and a specific component props could be defined as 

```tsx
type MyProps = StructuralEditorModeProps & {children: ReactNode, otherProp: unknown}
```